### PR TITLE
feat(entities): config store secret

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
# Summary

Integrate Konnect Vault with Konnect Config Store

KM-445

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/entities/entities-vaults/src/components/VaultForm.vue [hunk 1]
```diff
@@ -847,6 +849,7 @@ const updateFormValues = (data: Record<string, any>): void => {
   if (config && (Object.keys(config).length || data?.name === VaultProviders.KONNECT)) {
     vaultProvider.value = data?.item?.name || data?.name || ''
     originalVaultProvider.value = vaultProvider.value
+    configStoreId.value = data?.config_store_id || undefined
```
Comment (sumimakito) [inline packages/entities/entities-vaults/src/components/VaultForm.vue:852]: I encountered an issue where the secrets were gone after updating a Konnect vault. It seems the UI is creating a new `config-store` upon `vault`'s update:

https://github.com/user-attachments/assets/ff9070ed-5071-4eaf-b907-d1ab2cf5ceb3

After some quick investigation, `config_store_id` fields appear to be nested under different hierarchies in the responses of the following endpoints:

#### /runtime_groups/0c9a728b-8cf5-450c-97c3-995876d13495/v1/vaults/87ea53c8-2eea-4056-a3e7-934d07f7e210

Screenshot 2024-09-09 at 17 18 45

#### /v2/control-planes/0c9a728b-8cf5-450c-97c3-995876d13495/core-entities/vaults/87ea53c8-2eea-4056-a3e7-934d07f7e210

Screenshot 2024-09-09 at 17 20 39

I doubt that made the ref `configStoreId` here null-ish while updating a vault.

Comment (Leopoldthecoder) [reply packages/entities/entities-vaults/src/components/VaultForm.vue]: I pushed a [fix](https://github.com/Kong/public-ui-components/pull/1601/commits/a64a6e2f64c4b11e1dbad6f65013cfef4af81a64)

Comment (sumimakito) [reply packages/entities/entities-vaults/src/components/VaultForm.vue]: I can confirm that this is fixed. Thanks

File: packages/entities/entities-vaults/src/components/VaultList.vue [hunk 2]
```diff
@@ -402,6 +402,19 @@ const hideDeleteModal = (): void => {
...
+  const url = `${apiBaseUrl}${endpoints.list[app].deleteConfigStore}`
+    .replace(/{controlPlaneId}/gi, controlPlaneId || '')
+    .replace(/{id}/gi, configStoreId)
+  try {
+    await axiosInstance.delete(url)
+  } catch (error: any) {
+    // There is a rare case where more than 1 vaults are linked to the config store (i.e. created via API).
+    // In this case, the deletion will fail but the UI can safely ignore the error.
+  }
+}
```
Comment (sumimakito) [inline packages/entities/entities-vaults/src/components/VaultList.vue:416]: qq: Would we delete the associated config store if a user tries to delete a Konnect vault from the vault detail page as well?

**Review by sumimakito — APPROVED** [review]
Functionally tested the vault and secret picker features and LGTM. Left a non-blocking question.

File: packages/entities/entities-vaults/src/components/VaultList.vue (hunk 2 above)
Comment (Leopoldthecoder) [reply packages/entities/entities-vaults/src/components/VaultList.vue:416]: Yes I think we should. I'll add this logic in the host app.

**Pull request merged**